tests/common/tf/retrieval/retrieval_utils.py

```diff
@@ -0,0 +1,575 @@
+    schema_selected = schema.remove_by_tag(Tags.USER_ID)
+
+    embedding_dims = {}
+    if item_id_emb_size:
+        item_id_feature_name = schema.select_by_tag(Tags.ITEM_ID).column_names[0]
+        # item_id_domain = schema_utils.categorical_domains(schema)[item_id_feature_name]
+        embedding_dims[item_id_feature_name] = item_id_emb_size
+    else:
+        raise Exception(
+            "You must specify --item_id_emb_size for YouTubeDNN "
+            "as the last layer dim of the MLP must match the item id embedding dim."
+        )
+
+    emb_init = get_embeddings_initializer(emb_init_distr, emb_init_range)
+
+    layers_dims = []
+
+    layers_dims = list([int(v.strip()) for v in two_tower_mlp_layers.split(",") if v != ""])
+    # Appends the top MLP layer with the same size as the item id embedding
+    layers_dims.append(item_id_emb_size)
+
+    model = mm.YoutubeDNNRetrievalModel(
+        schema=schema_selected,
+        max_seq_length=max_seq_length,
+        num_sampled=youtubednn_sampled_softmax_n_candidates,
+        sampled_softmax=youtubednn_sampled_softmax,
+        logits_temperature=logits_temperature,
+        embedding_options=mm.EmbeddingOptions(
+            embedding_dims=embedding_dims,
+            infer_embedding_sizes=True,
+            infer_embedding_sizes_multiplier=two_tower_embedding_sizes_multiplier,
+            infer_embeddings_ensure_dim_multiple_of_8=True,
+            embeddings_initializers=emb_init,
+            embeddings_l2_reg=embeddings_l2_reg,
+        ),
+        top_block=mm.MLPBlock(
+            layers_dims,
+            activation=two_tower_activation,
+            kernel_initializer=two_tower_mlp_init,
+            no_activation_last_layer=True,
+            dropout=two_tower_dropout,
+            kernel_regularizer=regularizers.l2(l2_reg),
+            bias_regularizer=regularizers.l2(l2_reg),
+        ),
+        l2_normalization=cosine_similarity_logits,
+    )
+    return model
+
+
+def get_optimizer(
+    lr=1e-4,
+    lr_decay_rate=0.99,
+    lr_decay_steps=100,
+    optimizer="adam",
+    opt_clip_norm=None,
+    opt_clip_value=None,
+):
+    lerning_rate = lr
+    if lr_decay_rate:
+        lerning_rate = tf.keras.optimizers.schedules.ExponentialDecay(
+            lr,
+            decay_steps=lr_decay_steps,
+            decay_rate=lr_decay_rate,
+            staircase=True,
+        )
+
+    if optimizer == "adam":
+        opt = tf.keras.optimizers.Adam(
+            learning_rate=lerning_rate,
+            clipnorm=opt_clip_norm,
+            clipvalue=opt_clip_value,
+        )
+    elif optimizer == "adagrad":
+        opt = tf.keras.optimizers.Adagrad(
+            learning_rate=lerning_rate,
+            clipnorm=opt_clip_norm,
+            clipvalue=opt_clip_value,
+        )
+    else:
+        raise ValueError("Invalid optimizer")
+
+    return opt
+
+
+class ExamplesPerSecondCallback(tf.keras.callbacks.Callback):
```

Comment:
Should we move this to merlin-models? Might be useful outside of these tests as well.

Reply:
@marcromeyn  The `ExamplesPerSecondCallback` depends on `wandb` library. It is currently required only when running the integration tests. I have added the `wandb` only as dev requirement, so that CI and developers that want to check the metrics of the retrieval integration tests can do so. 
I think that maybe in the future we might having a logging module for Merlin Models, where me provide callbacks and utils for logging tools like Tensorboard, W&B, MLFlow. Then maybe then we could move this `ExamplesPerSecondCallback` to such logging module.
Does that make sense to you?

Reply:
I think we can remove that dependency but we can do that in a future PR